File: app/src/main/cpp/model/Calculator.cpp
//
// Created by kiber_god on 11.09.2023.
//
#include <cstring>
#include "Calculator.h"

Calculator::Calculator() {
    expression[0] = '\0';
}

Calculator::Calculator(const char* expression) {
    strncpy(this->expression, expression, MAX_NUMBER_LENGTH - 1);
    this->expression[MAX_NUMBER_LENGTH - 1] = '\0';
}

const char* Calculator::getExpression() const {
    return expression;
}

File: app/src/main/java/com/kibergod/passwordstorage/model/Calculator.java
package com.kibergod.passwordstorage.model;

/*
 *   Даний клас містить певні дані, пов`язані з калькулятором та не повинен
 *   існувати більш ніж в одному екземплярі
 */
public class Calculator {
    private String expression;

    public Calculator(String expression) {
        this.expression = expression;
    }

    public String getExpression() { return expression; }
}
